### ISSUE TITLE
Session time-outs: need to allow for session extension (accessibility/508)

### ISSUE THREAD

**Issue by iamjolly** [18F/tock#1181]
When sessions time-outs occur, no warning or ability to extend a user's session is offered. This leads to usability and accessibility problems.

Per [WCAG 2.2.1 Timing Adjustable (Level A)](https://www.w3.org/TR/UNDERSTANDING-WCAG20/time-limits-required-behaviors.html):

For each time limit that is set by the content, at least one of the following is true: 
- **Turn off:** The user is allowed to turn off the time limit before encountering it; or
- **Adjust:** The user is allowed to adjust the time limit before encountering it over a wide range that is at least ten times the length of the default setting; or
- **Extend:** The user is warned before time expires and given at least 20 seconds to extend the time limit with a simple action (for example, "press the space bar"), and the user is allowed to extend the time limit at least ten times; or
- **Real-time Exception:** The time limit is a required part of a real-time event (for example, an auction), and no alternative to the time limit is possible; or
- **Essential Exception:** The time limit is essential and extending it would invalidate the activity; or
- **20 Hour Exception:** The time limit is longer than 20 hours.

This will also help users who do not require accessibility supports but may be frustrated by not realizing a session is "dead" and not being able to log new entries (losing any unsaved entries) due to needed to log back in. 

**Recommendation:** 
1. Prompt users with a modal dialog with a warning that lack of activity may require log-out and offer to allow them to extend their session. 
2. If a user is logged out for inactivity, redirect them to a logged-out page with a link to log back in.

**Comment by Jkrzy** [comment]
Initial pass at this implemented in #1216!

**Comment by Jkrzy** [comment]
As-is this uses the default alert dialog, we'll open an issue to iterate onward.

> Doing this with a USWDS confirmation modal will give us a bit more control and let us make this even more accessible. Replacing the alert dialog with something where we can better control the title and interpretation by a screenreader.
> 
> I'm hopeful there's some existing art we can borrow from another TTS site
